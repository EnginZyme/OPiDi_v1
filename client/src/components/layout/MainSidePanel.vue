<template>
  <div>
    <div class="function_nav_bar">
      <div class="function_nav_bar__grid">
        <img alt="EnginZyme Logo" src="../../assets/logo_light.svg" />
      </div>
      <div></div>
      <div class="function_nav_bar__grid">
        <UserAvatarMenu />
      </div>
    </div>
  </div>
</template>

<script>
import UserAvatarMenu from "../base/UserAvatarMenu";

/**
 * A component that renders the funciton navigation bar.
 */
export default {
  name: "MainSidePanel",
  components: {
    UserAvatarMenu,
  },
};
</script>

<style scoped>
.function_nav_bar__grid {
  display: grid;
  flex-direction: column;
  justify-items: center;
  align-self: center;
  grid-gap: 12px;
}
.function_nav_bar {
  background: #171a2e;
  height: 100%;
  width: 75px;
  display: grid;
  direction: column;
  grid-template-rows: 50px 1fr 70px;
  filter: drop-shadow(2px 2px 2px #040716);
}
</style>
