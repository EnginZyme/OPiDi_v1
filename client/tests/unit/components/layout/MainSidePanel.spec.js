import { shallowMount } from "@vue/test-utils"
import MainSidePanel from "@/components/layout/MainSidePanel"


describe("shallowMounted MainSidePanel", () => {
    const wrapper = shallowMount(MainSidePanel)

    it("renders correctly", () => {
        expect(wrapper.element).toMatchSnapshot();
    })
})
