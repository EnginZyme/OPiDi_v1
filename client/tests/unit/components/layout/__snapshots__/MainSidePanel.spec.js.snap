// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallowMounted MainSidePanel renders correctly 1`] = `
<div>
  <div
    class="function_nav_bar"
  >
    <div
      class="function_nav_bar__grid"
    >
      <img
        alt="EnginZyme Logo"
        src="../../assets/logo_light.svg"
      />
    </div>
     
    <div />
     
    <div
      class="function_nav_bar__grid"
    >
      <useravatarmenu-stub />
    </div>
  </div>
</div>
`;
